src/ripple/basics/Buffer.h

```diff
@@ -58,33 +93,28 @@ class Buffer
     */
     Buffer& operator= (Buffer&& other)
     {
-        p_ = std::move(other.p_);
-        size_ = other.size_;
-        other.size_ = 0;
+        if (this != &other)
+        {
+            p_ = std::move(other.p_);
+            size_ = other.size_;
+            other.size_ = 0;
+        }
         return *this;
     }
 
-    /** Create an uninitialized buffer with the given size. */
+    /** Construct from a slice */
     explicit
-    Buffer (std::size_t size)
-        : p_ (size ?
-            new std::uint8_t[size] : nullptr)
-        , size_ (size)
-    {
-    }
-
-    /** Create a buffer as a copy of existing memory. */
-    Buffer (void const* data, std::size_t size)
-        : Buffer (size)
+    Buffer (Slice s)
+        : Buffer (s.data(), s.size())
     {
-        std::memcpy(p_.get(), data, size);
     }
 
-    /** Create a buffer from a copy of existing memory. */
-    explicit
-    Buffer (Slice const& slice)
-        : Buffer(slice.data(), slice.size())
+    /** Assign from slice */
+    Buffer& operator= (Slice s)
```

Comment:
`std::string` should use parenthesis to construct, not braces. Its one of the exceptions (because it has a constructor that takes a `std::initializer_list`). `Slice` on the other hand, should construct using braces.